cli/src/pcluster/templates/cw_dashboard_builder.py

```diff
@@ -219,6 +229,116 @@ def _add_conditional_storage_widgets(
                 widgets_list.append(graph_widget)
         return widgets_list
 
+    def custom_pcluster_metric_filter(self, metric_name, filter_pattern, custom_namespace):
+        """Adding custom metric filter from named tuple."""
+        metric_filter = logs.CfnMetricFilter(
+            scope=self.stack_scope,
+            id=metric_name + " Filter",
+            filter_pattern=filter_pattern,
+            log_group_name=self.cw_log_group_name,
+            metric_transformations=[
+                logs.CfnMetricFilter.MetricTransformationProperty(
+                    metric_namespace=custom_namespace,
+                    metric_name=metric_name,
+                    metric_value="100",
+                    default_value=0,
+                )
+            ],
+        )
+        metric_filter.add_depends_on(self.cw_log_group)
+        return metric_filter
+
+    def add_custom_error_metrics(self):
```

Comment:
Shall we add a general error metric?
```
custom_metric_filter(
                metric_name="All errors",
                filter_pattern="?error ?Error ?ERROR",
            )
```

Reply:
I was talking to maurizo and that might be useful, wonder where to add it tho, also prob could add WARNING